projects/core/src/site-context/store/effects/currencies.effect.ts

```diff
@@ -41,12 +52,25 @@ export class CurrenciesEffects {
         this.winRef.sessionStorage.setItem('currency', action.payload);
       }
     }),
-    map(() => new SiteContextActions.CurrencyChange())
+    withLatestFrom(
+      this.state.select(getActiveCurrency).pipe(
+        bufferCount(2, 1),
+        // avoid dispatching `change` action when we're just setting the initial value:
+        filter(([previous]) => !!previous),
+        // avoid dispatching `change` action when values are the same
+        filter(([previous, current]) => previous !== current)
+      )
+    ),
+    map(
+      ([_, [previous, current]]) =>
+        new SiteContextActions.CurrencyChange({ previous, current })
```

Comment:
If "currency change" effect is just observing the state, why not derive it from state directly?